src/ClientHandle.cpp

```diff
@@ -1863,6 +1863,12 @@ void cClientHandle::RemoveFromWorld(void)
 	m_LastStreamedChunkZ = 0x7fffffff;
 
 	m_HasSentPlayerChunk = false;
+
+	// Do not send chunks for 1.5 seconds.
+	// Apparently, the chunks of the old world fail to unload if
+	// chunks in the new world which have the same coords are sent too fast after the unload.
+	// this is a temp fix, find the root cause!
+	m_ChunkSendDelay = 30;
```

Comment:
Well...  The client's supposed to unload the chunks when it changes dimensions or when respawning.  However, there _is_ a special case, which occurs with multiworld, regarding the [respawn packet](http://wiki.vg/Protocol#Respawn), given that players neither die nor respawn:

> <h3>Respawn</h3>
> 
> To change the player's dimension (overworld/nether/end), send them a respawn packet with the appropriate dimension, followed by prechunks/chunks for the new dimension, and finally a position and look packet. You do not need to unload chunks, the client will do it automatically. 
> 
> > **Warning**: Avoid changing player's dimension to same dimension they were already in unless they are dead. If you change the dimension to one they are already in, weird bugs can occur, such as the player being unable to attack other players in new world (until they die and respawn). 
> 
> > If you must respawn a player in the same dimension without killing them, send two respawn packets, one to a different world and then another to the world you want. You do not need to complete the first respawn; it only matters that you send two packets. 

Code-wise, the client only unloads chunks and resets the world if the dimension changes.  From `NetHandlerPlayClient`:

```java
    @Override
    public void handleRespawn(S07PacketRespawn packetIn) {
        PacketThreadUtil.checkThreadAndEnqueue(packetIn, this, this.gameController);

        if (packetIn.getDimensionID() != this.gameController.thePlayer.dimension) {
            this.doneLoadingTerrain = false;
            Scoreboard var2 = this.clientWorldController.getScoreboard();
            this.clientWorldController = new WorldClient(this, new WorldSettings(0L, packetIn.getGameType(), false, this.gameController.theWorld.getWorldInfo().isHardcoreModeEnabled(), packetIn.getWorldType()), packetIn.getWorldType(), packetIn.getDifficulty(), this.gameController.mcProfiler);
            this.clientWorldController.setWorldScoreboard(var2);
            this.gameController.loadWorld(this.clientWorldController);
            this.gameController.thePlayer.dimension = packetIn.getWorldType();
            this.gameController.displayGuiScreen(new GuiDownloadTerrain(this));
        }

        this.gameController
        .setDimensionAndSpawnPlayer(packetIn.getWorldType());
        this.gameController.playerController.setGameType(packetIn.getGameType());
    }
```